### PR TITLE
feat(library): add PolicyAI Integration for Content Moderation

### PR DESCRIPTION
**Pull request by vitchor**
### Summary

This PR adds [PolicyAI](https://musubilabs.ai) by Musubi Labs as an out-of-the-box input and output rail for NeMo Guardrails. PolicyAI enables teams to enforce custom content policies on LLM inputs and outputs, providing flexible guardrails that adapt to your specific use cases rather than relying on fixed, generic moderation categories.

### What is PolicyAI?

PolicyAI is a policy-based content moderation API built for LLM applications. Unlike traditional moderation tools with rigid categories, PolicyAI lets you:

- **Define custom policies in natural language** — Create policies tailored to your domain (e.g., "block unauthorized refund promises", "prevent disclosure of internal pricing")
- **Organize policies with tags** — Group policies by environment (`prod`, `staging`, `dev`) or use case, making it easy to test and iterate
- **Cluster & explore content to detect novel emerging attacks** — Discover patterns in flagged content to proactively identify new threats before they become widespread
- **Get structured decisions** — Receive assessment (`SAFE`/`UNSAFE`), category, severity (0-3), and reasoning for each evaluation

### Changes

- **`nemoguardrails/library/policyai/`** — New integration module
  - `__init__.py` — Package initialization
  - `actions.py` — `call_policyai_api` action that calls the `/policyai/v1/decisions/evaluate/{tag_name}` endpoint
  - `flows.co` — Colang 2.x flows for input/output moderation
  - `flows.v1.co` — Colang 1.x flows for input/output moderation
- **`docs/user-guides/community/policyai.md`** — User documentation
- **`docs/configure-rails/guardrail-catalog.md`** — Added PolicyAI to Third-Party APIs section
- **`tests/test_policyai_rail.py`** — Unit tests with mocked API responses

### Usage

```yaml
models:
  - type: main
    engine: openai
    model: gpt-4

rails:
  input:
    flows:
      - policyai moderation on input
  output:
    flows:
      - policyai moderation on output
```

### Environment Variables

| Variable | Required | Default | Description |
|----------|----------|---------|-------------|
| `POLICYAI_API_KEY` | Yes | — | Your PolicyAI API key (sent as `Musubi-Api-Key` header) |
| `POLICYAI_BASE_URL` | No | `https://api.musubilabs.ai` | PolicyAI API base URL |
| `POLICYAI_TAG_NAME` | No | `prod` | Policy tag to use for evaluation |

### How It Works

1. When a message is sent (user input or bot output), the integration calls PolicyAI's tag-based evaluation endpoint
2. PolicyAI evaluates the content against **all policies attached to the specified tag**
3. If **any** policy returns `UNSAFE`, the message is blocked and replaced with a refusal response
4. Supports `enable_rails_exceptions` config for custom exception handling
5. **Fail-closed behavior**: Raises an error if no policies are attached to the tag or all evaluations fail
6. HTTP requests have a **30-second timeout** to prevent hanging

### API Response Handling

The integration aggregates results from multiple policies:
- Returns the first `UNSAFE` result found (with category, severity, and reason)
- Returns `SAFE` only if all policies pass
- Gracefully handles failed policy evaluations

### Testing

Comprehensive test coverage using mocked API responses via `aioresponses`, including:
- Input/output moderation (safe and unsafe cases)
- Custom policy tags via environment variable
- Multiple policy evaluation
- Edge cases: empty data arrays, API errors, all policies failed, missing API key
- Default value handling and URL normalization

### Reference

This integration follows the same pattern as the existing ActiveFence integration (`nemoguardrails/library/activefence/`).

### Links

- [PolicyAI Documentation](https://docs.musubilabs.ai)
- [Musubi Labs](https://musubilabs.ai)

### CODE REVIEW

File: nemoguardrails/library/policyai/flows.co [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  if $result.assessment == "UNSAFE"
+    if $system.config.enable_rails_exceptions
+      send PolicyAIModerationRailException(message="PolicyAI moderation triggered. Content violated policy: " + $result.category)
+    else
+      bot refuse to respond
+    abort
+
+flow bot inform policyai violation
+  bot say "I cannot process this request as it violates our content policies."
```
Comment (Pouyanpi) [inline nemoguardrails/library/policyai/flows.co]: this flow is not used? do you intend to use it at line 32 instead?

File: nemoguardrails/library/policyai/flows.v1.co [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  if $result.assessment == "UNSAFE"
+    if $config.enable_rails_exceptions
+      create event PolicyAIModerationRailException(message=$result.exception_message)
+    else
+      bot refuse to respond
+    stop
+
+define bot inform policyai violation
+  "I cannot process this request as it violates our content policies."
```
Comment (Pouyanpi) [inline nemoguardrails/library/policyai/flows.v1.co]: same as above: 

this flow is not used? do you intend to use it at line 32 instead?

File: nemoguardrails/library/policyai/flows.co (hunk 1 above)
Comment (vitchor) [reply nemoguardrails/library/policyai/flows.co]: @Pouyanpi – Good catch! I originally added this as a placeholder thinking it might be useful for customization, but you're right — it's not being used. I've removed it to keep things clean. Thanks for the review!

File: nemoguardrails/library/policyai/flows.v1.co (hunk 2 above)
Comment (vitchor) [reply nemoguardrails/library/policyai/flows.v1.co]: same as above:

@Pouyanpi – Good catch! I originally added this as a placeholder thinking it might be useful for customization, but you're right — it's not being used. I've removed it to keep things clean. Thanks for the review!

**Review by Pouyanpi — APPROVED** [review]
Thank you @vitchor for the PR. It looks good to merge after gpg signing your commits 👍🏻

**Comment by vitchor** [comment]
> Thank you @vitchor for the PR. It looks good to merge after gpg signing your commits 👍🏻

Thanks so much for the support @Pouyanpi 🎉 👏  I've signed all the commits and will be merging this today.

Thanks again and hope you have a great week! 😄

**Comment by vitchor** [comment]
Oh, actually @Pouyanpi – I've realized I can't merge it myself 😄  . Could you please merge it? The commits have been signed. ✅  Thank you!

**Comment by Pouyanpi** [comment]
@vitchor sorry for the late reply! I overlooked your last comment. Could you please rebase on develop and resolve the conflicts? The conflicts are related to documentation changes. Once that’s done, I’ll merge it.

It will be included in the 0.21.0 release.

**Comment by vitchor** [comment]
Hi, @Pouyanpi – no worries and thanks for the follow-up 😄 

I've already rebased on `develop` and fixed the conflict. ✅ 

I'm looking forward to having it live on the `0.21.0` release – thanks again! 🎉

**Comment by vitchor** [comment]
BTW, I had to rebase again to fix a Sphinx build warning — a code block in `policyai.md` was using an unrecognized `colang` Pygments lexer, which fails the strict (`-W`) docs build.

```diff
# docs/configure-rails/guardrail-catalog/community/policyai.md:88
-```colang
+```text
```

**Comment by Pouyanpi** [comment]
Thank you @vitchor for your collaboration! Merged 🚀

**Comment by vitchor** [comment]
@Pouyanpi – awesome! Btw, do you have any estimates on when `0.21.0` will be released? Thanks!
